ouroboros-consensus/src/Ouroboros/Consensus/Demo/Ledger/Byron/Forge.hs

```diff
@@ -32,24 +33,75 @@ import           Ouroboros.Consensus.Ledger.Byron
 import           Ouroboros.Consensus.Protocol.Abstract
 import           Ouroboros.Consensus.Protocol.ExtNodeConfig
 import           Ouroboros.Consensus.Protocol.PBFT
+import           Ouroboros.Consensus.Protocol.WithEBBs
 
 import           Ouroboros.Consensus.Demo.Ledger.Byron.Config
 
+forgeBlockOrEBB
+  :: forall m cfg.
+     ( HasNodeState_ () m  -- @()@ is the @NodeState@ of PBFT
+     , MonadRandom m
+       -- TODO: This 'Given' constraint should not be needed (present in config)
+     , Given Crypto.ProtocolMagicId
+     )
+  => NodeConfig ByronEBBExtNodeConfig
+  -> SlotNo                       -- ^ Current slot
+  -> BlockNo                      -- ^ Current block number
+  -> ChainHash (ByronBlockOrEBB cfg)   -- ^ Previous hash
+  -> [GenTx (ByronBlockOrEBB cfg)]     -- ^ Txs to add in the block
+  -> ()                           -- ^ Leader proof ('IsLeader')
+  -> m (ByronBlockOrEBB ByronDemoConfig)
+forgeBlockOrEBB cfg curSlot curNo prevHash txs () = case prevHash of
+  GenesisHash -> forgeGenesisEBB cfg curSlot
```

Comment:
This means that only the very first block (the one after genesis) produced will be an EBB. I thought that an EBB was produced at the start of each epoch?

Reply:
Yes, but this isn't needed for the demo. We only need a block at the actual genesis to distinguish the genesis hash.